controller/iteration_blackbox_test.go

```diff
@@ -1080,3 +1080,193 @@ func (rest *TestIterationREST) TestIterationDelete() {
+		// 	|---- itr 6
+
+		// when
+		children, err := rest.db.Iterations().LoadChildren(svc.Context, itr2.ID)
+		// then
+		require.Nil(t, err)
+		require.Len(t, children, 0)
+
+		// when
+		children, err = rest.db.Iterations().LoadChildren(svc.Context, itr1.ID)
+		// then
+		require.Nil(t, err)
+		require.Len(t, children, 4)
+
+		allChildren := map[uuid.UUID]struct{}{
+			// expected subtree of itr 1
+			itr2.ID: {},
+			itr3.ID: {},
+			itr4.ID: {},
+			itr5.ID: {},
+		}
+		for _, i := range children {
+			delete(allChildren, i.ID)
+		}
+		require.Empty(t, allChildren)
+
+		// when
+		children, err = rest.db.Iterations().LoadChildren(svc.Context, itr3.ID)
+		// then
+		require.Nil(t, err)
+		require.Len(t, children, 1)
+
+		allChildren = map[uuid.UUID]struct{}{
+			itr4.ID: {},
+		}
+		for _, i := range children {
+			delete(allChildren, i.ID)
+		}
+		require.Empty(t, allChildren)
+	})
+
+	rest.T().Run("update fail - parent of root iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		typeIterationString := iteration.APIStringTypeIteration
+		rootItr := fxt.Iterations[0]
+		itr1 := fxt.Iterations[1]
+		newParentIDStr := itr1.ID.String()
+		payload := app.UpdateIterationPayload{
+			Data: &app.Iteration{
+				Attributes: &app.IterationAttributes{},
+				Relationships: &app.IterationRelations{
+					Parent: &app.RelationGeneric{
+						Data: &app.GenericData{
+							ID:   &newParentIDStr,
+							Type: &typeIterationString,
+						},
+					},
+				},
+				ID:   &rootItr.ID,
+				Type: typeIterationString,
+			},
+		}
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		test.UpdateIterationForbidden(t, svc.Context, svc, ctrl, rootItr.ID.String(), &payload)
+	})
+
+	rest.T().Run("update fail - non-existing parent of iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		typeIterationString := iteration.APIStringTypeIteration
+		itr1 := fxt.Iterations[1]
+		newParentIDStr := "73048351-a4c0-4cf6-aff9-1fa3d7576ac0"
+		payload := app.UpdateIterationPayload{
+			Data: &app.Iteration{
+				Attributes: &app.IterationAttributes{},
+				Relationships: &app.IterationRelations{
+					Parent: &app.RelationGeneric{
+						Data: &app.GenericData{
+							ID:   &newParentIDStr,
+							Type: &typeIterationString,
+						},
+					},
+				},
+				ID:   &itr1.ID,
+				Type: typeIterationString,
+			},
+		}
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		test.UpdateIterationNotFound(t, svc.Context, svc, ctrl, itr1.ID.String(), &payload)
+	})
+
+	rest.T().Run("update fail - invalid UUID parent of iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		typeIterationString := iteration.APIStringTypeIteration
+		itr1 := fxt.Iterations[1]
+		newParentIDStr := "/"
+		payload := app.UpdateIterationPayload{
```

Comment:
I would make a function out of this repetitive payload creation. I've seen it for at least 3 times, if not more.

Reply:
yes now I have added a function to do the same https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/75622623f469327dd41526bd841df5dae623e6a2